File: backend/app/schemas/bid.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class BidCreate(BaseModel):
    tender_id: str
    stage: str = "new"
    notes: str = ""


class BidUpdate(BaseModel):
    stage: Optional[str] = None
    notes: Optional[str] = None


class BidResponse(BaseModel):
    id: str
    user_id: str
    tender_id: str
    stage: str
    match_score: Optional[float] = None
    notes: str
    created_at: datetime
    updated_at: datetime

    # Tender details (joined)
    tender_title: Optional[str] = None
    tender_authority: Optional[str] = None
    tender_deadline: Optional[datetime] = None
    tender_budget_raw: Optional[str] = None
    tender_source: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/types/bid.ts
export interface Bid {
  id: string;
  user_id: string;
  tender_id: string;
  stage: string;
  match_score: number | null;
  notes: string;
  created_at: string;
  updated_at: string;
  tender_title: string | null;
  tender_authority: string | null;
  tender_deadline: string | null;
  tender_budget_raw: string | null;
  tender_source: string | null;
}

export const STAGES = [
  { id: "new", label: "New", color: "bg-gray-100" },
  { id: "interested", label: "Interested", color: "bg-blue-50" },
  { id: "evaluating", label: "Evaluating", color: "bg-purple-50" },
  { id: "drafting", label: "Drafting", color: "bg-amber-50" },
  { id: "submitted", label: "Submitted", color: "bg-orange-50" },
  { id: "won", label: "Won", color: "bg-green-50" },
  { id: "lost", label: "Lost", color: "bg-red-50" },
] as const;